src/utils.ts

```diff
@@ -129,16 +134,72 @@ export const assertEOSError = async (
 				error.json.error.name === eosErrorName,
 				`Expected ${eosErrorName}, got ${error.json.error.name} instead.`
 			);
-			return;
+			if (furtherErrorCheck) {
+				furtherErrorCheck(error)
+			};
 		} else {
 			// Fail if error not thrown by EOS
 			assert.fail(
 				`Expected EOS error ${eosErrorName}, but got ${JSON.stringify(error, null, 4)} instead.`
 			);
 		}
+		return true
+	}
+	return false;
+};
+
+/**
+ * Asserts EOS throws an error and validates the error output name matches the expected `eosErrorName`
+ * @author Kevin Brown <github.com/thekevinbrown>
+ * @param operation Operation promise
+ * @param eosErrorName Expected EOS error name
+ * @param description Output message description
+ */
+export const assertEOSError = async (
+	operation: Promise<any>,
+	eosErrorName: string,
+	description: string
+) => {
+	if (assertExpectedEOSError(operation, eosErrorName, (_) => {})) {
+		// Execute operation and handle exceptions
+		assert.fail(`Expected ${description} but operation completed successfully.`);
+	}
+};
+
+/**
+ * Asserts EOS throws an error and validates the error output name matches the
+ * expected 'eosio_assert_message_exception' and the error message includes `description`
+ * @author Dallas Johnson <github.com/dallasjohnson>
+ * @param operation Operation promise
+ * @param message Output message expected to be included
+ */
+export const assertEOSErrorIncludesMessage = async (operation: Promise<any>, message: string) => {
+	const eosErrorName = 'eosio_assert_message_exception';
+	// Execute operation and handle exceptions
+	console.log('should get here for the error' + message);
+
+	if (
+		!(await assertExpectedEOSError(operation, eosErrorName, (error)  => {
+			let errorMessage = error.json.error.details[0].message;
+			console.log('should get here before error meassge chack: ' + JSON.stringify(error));
```

Comment:
Would be good to remove this logging as I think it's a remnant of debugging while working.

Reply:
👍 🤦‍♂